evap/static/ts/src/csrf-utils.ts

```diff
@@ -28,6 +32,27 @@ $.ajaxSetup({
     },
 });
 
+function addCsrfToken(this: HTMLFormElement, ev: Event) {
+    const inputElement = document.createElement("input");
+    inputElement.setAttribute("type", "hidden");
+    inputElement.setAttribute("name", "csrfmiddlewaretoken");
+    inputElement.setAttribute("value", csrftoken);
+
+    const { submitter } = ev as SubmitEvent;
+    const action = (submitter && submitter.getAttribute("formaction")) || this.getAttribute("action");
+    const method = (submitter && submitter.getAttribute("formmethod")) || this.getAttribute("method");
+
+    if (method && !isMethodCsrfSafe(method) && action && !isUrlCrossDomain(action)) {
+        this.insertAdjacentElement("afterbegin", inputElement.cloneNode() as Element);
+    }
+}
+
+// Add CSRF token to regular forms on submit.
+// This reduces boilerplate code in the templates and makes the form's HTML cacheable.
+for (const form of document.forms) {
+    form.addEventListener("submit", addCsrfToken);
```

Comment:
I think it would be easier to inline the function definition here and use `form` instead of `this` (it took me a minute to find out what the syntax for `this` type hints means :D)